pm: cannot build gforker and remshell

All my builds using `--with-pm=gforker` are broken: https://github.com/mpi4py/mpi4py-testing/actions/runs/2139048195

For example, look this log: https://github.com/mpi4py/mpi4py-testing/runs/5951439920?check_suite_focus=true

```
  CC       src/pm/gforker/src_pm_gforker_mpiexec-mpiexec.o
src/pm/gforker/mpiexec.c:84:10: fatal error: simple_pmiutil.h: No such file or directory
   84 | #include "simple_pmiutil.h"
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [Makefile:30127: src/pm/gforker/src_pm_gforker_mpiexec-mpiexec.o] Error 1
```

In addition, you can see the following unrelated format warnings:

```
In file included from /home/runner/work/mpi4py-testing/mpi4py-testing/src/mpl/include/mpl_argstr.h:14,
                 from /home/runner/work/mpi4py-testing/mpi4py-testing/src/mpl/include/mpl.h:13,
                 from ./src/include/mpiimpl.h:119,
                 from src/nameserv/file/file_nameserv.c:14:
./src/mpid/ch4/shm/src/../posix/posix_coll_release_gather.h: In function ‘MPIDI_POSIX_mpi_allreduce_release_gather’:
./src/mpid/ch4/shm/src/../posix/posix_coll_release_gather.h:352:46: warning: format ‘%d’ expects argument of type ‘int’, but argument 4 has type ‘uint64_t’ {aka ‘long unsigned int’} [-Wformat=]
  352 |                                              "Reduce shmgr pipeline info: segsize=%d count="
      |                                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/runner/work/mpi4py-testing/mpi4py-testing/src/mpl/include/mpl_dbg.h:73:37: note: in definition of macro ‘MPL_DBG_MSG_FMT’
   73 |             int _ret = MPL_snprintf _fmatargs ;                          \
      |                                     ^~~~~~~~~
In file included from ./src/mpid/ch4/shm/src/../posix/posix_coll.h:11,
                 from ./src/mpid/ch4/shm/src/../posix/shm_inline.h:16,
                 from ./src/mpid/ch4/shm/src/shm_coll.h:10,
                 from ./src/mpid/ch4/shm/src/shm_impl.h:18,
                 from ./src/mpid/ch4/include/mpidch4.h:343,
                 from ./src/mpid/ch4/include/mpidpost.h:10,
                 from ./src/include/mpiimpl.h:230,
                 from src/nameserv/file/file_nameserv.c:14:
./src/mpid/ch4/shm/src/../posix/posix_coll_release_gather.h:352:84: note: format string is defined here
  352 |                                              "Reduce shmgr pipeline info: segsize=%d count="
      |                                                                                   ~^
      |                                                                                    |
      |                                                                                    int
      |                                                                                   %ld
```

--- a/src/pm/util/Makefile.mk
+++ b/src/pm/util/Makefile.mk
@@ -7,7 +7,6 @@
 # managers that use utility code from this directory
 common_pm_includes = \
     -I${top_builddir}/src/include -I${top_srcdir}/src/include \
-    -I${top_builddir}/src/pmi/simple -I${top_srcdir}/src/pmi/simple \
     -I${top_builddir}/src/pm/util -I${top_srcdir}/src/pm/util
 
 if BUILD_PM_UTIL

--- a/src/pm/util/simple_pmiutil.h
+++ b/src/pm/util/simple_pmiutil.h
@@ -1,0 +1,37 @@
+/*
+ * Copyright (C) by Argonne National Laboratory
+ *     See COPYRIGHT in top-level directory
+ */
+
+#ifndef SIMPLE_PMIUTIL_H_INCLUDED
+#define SIMPLE_PMIUTIL_H_INCLUDED
+
+/* maximum sizes for arrays */
+#define PMIU_MAXLINE 1024
+#define PMIU_IDSIZE    32
+
+/* we don't have access to MPIR_Assert and friends here in the PMI code */
+#if defined(HAVE_ASSERT_H)
+#include <assert.h>
+#define PMIU_Assert(expr) assert(expr)
+#else
+#define PMIU_Assert(expr)
+#endif
+
+#if defined HAVE_ARPA_INET_H
+#include <arpa/inet.h>
+#endif /* HAVE_ARPA_INET_H */
+
+
+/* prototypes for PMIU routines */
+void PMIU_Set_rank(int PMI_rank);
+void PMIU_SetServer(void);
+void PMIU_printf(int print_flag, const char *fmt, ...);
+int PMIU_readline(int fd, char *buf, int max);
+int PMIU_writeline(int fd, char *buf);
+int PMIU_parse_keyvals(char *st);
+void PMIU_dump_keyvals(void);
+char *PMIU_getval(const char *keystr, char *valstr, int vallen);
+void PMIU_chgval(const char *keystr, char *valstr);
+
+#endif /* SIMPLE_PMIUTIL_H_INCLUDED */

